### PR TITLE
make the hack/local-up-cluster.sh cni plugin installation working on linux arm64

### PR DESCRIPTION
**Pull request by demko**
#### What type of PR is this?

/kind cleanup

#### What this PR does / why we need it:

Currently  the `hack/local-up-cluster.sh` cni plugin installation has `amd64` hard-coded in a couple of places.
Running `hack/local-up-cluster.sh` on `arm64` results in installing the cni plugin for the wrong architecture (arm64 kubelet with an amd64 cni plugin)
This PR is removing any hard-coded reference to `amd64` and reusing `CNI_TARGETARCH` as needed, so running 

```
CNI_TARGETARCH=arm64 ./hack/local-up-cluster.sh
```
will honour the `CNI_TARGETARCH` value during the CNI installation process

#### Which issue(s) this PR fixes:

Fixes #

#### Special notes for your reviewer:

A couple of notes on the implementation:
- on purpose made as the smallest possible change
- not addressing some corner cases 
  - e.g. CNI plugin already installed for the wrong architecture is preventing the right one to be installed
- leveraging some specific `bash` functionalities
  - dynamic variables: ${!variable}
  - uppercase conversion: ${variable^^}
#### Does this PR introduce a user-facing change?

```release-note
None
```

#### Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by dims** [comment]
@demko etcd is also amd64 by default. we'll need to wait until 3.6.0 is released

**Comment by adisky** [comment]
/ok-to-test
@dims CNI is also default amd64,  Its hardcoded at one place @demko is trying to fix that

File: hack/local-up-cluster.sh [hunk 1]
```diff
@@ -1043,7 +1043,8 @@ function parse_eviction {
 function install_cni {
   echo "Installing CNI plugin binaries ..." \
     && curl -sSL --retry 5 --output /tmp/cni."${CNI_TARGETARCH}".tgz "${CNI_PLUGINS_URL}" \
-    && echo "${CNI_PLUGINS_AMD64_SHA256SUM}  /tmp/cni.amd64.tgz" | tee /tmp/cni.sha256 \
+    && CNI_PLUGINS_SHA256SUM="CNI_PLUGINS_${CNI_TARGETARCH^^}_SHA256SUM" \
```
Comment (adisky) [inline hack/local-up-cluster.sh:1046]: you can follow the kind scripts to fix this
https://github.com/kubernetes-sigs/kind/blob/c45ee4a90254d86c9e06c227cc0ac0f557e8f9c8/images/base/Dockerfile#L173-#L177

**Comment by demko** [comment]
@dims the etcd installation in `hack/install_etcd.sh` seems to automatically detect the architecture and install the correct binaries
```
{"level":"info",...,"embed/etcd.go:307"
,"msg":"starting an etcd server"
,"etcd-version":"3.5.1"
,"git-sha":"e8732fb5f"
,"go-version":"go1.16.3"
,"go-os":"linux"
,"go-arch":"arm64"...}
```

What is preventing me to have `local-up-cluster.sh` working on linux arm64 (this is by running Ubuntu 21.10 on an Apple M1 using UTM)

**Comment by dims** [comment]
@adisky ack. please assign this to me when ready.

**Pull request closed without merging**